### ISSUE TITLE
Problems with Vulkan/DXGI on AMD GPU

### ISSUE THREAD

**Issue by SaltyBet** [SpecialKO/SpecialK#208]
According to these:
- [DXGI Swapchain on Opengl and Vulkan Games](https://community.amd.com/t5/opengl-vulkan/dxgi-swapchain-on-opengl-and-vulkan-games/m-p/650623/highlight/true)
- https://github.com/SpecialKO/SpecialK/issues/141

... I understand Vulkan/DXGI interop should work on AMD?

Tested RPCS3 using Vulkan and colors are broken when trying to enable HDR.

![Monitor output](https://github.com/user-attachments/assets/3525b194-d405-4b33-a31d-1df7100481b8)

### Logs:

[logs.zip](https://github.com/user-attachments/files/17167399/logs.zip)

**Comment by Kaldaien** [comment]
This is a known problem, unfortunately.
Special K fully works around their driver interop in OpenGL games.

In the Vulkan case, what's happening is SK is changing the D3D12 SwapChain format, and through a mechanism I don't know, they're copying the Vulkan image over to the D3D12 SwapChain without accounting for format mismatch.

It's difficult for me to develop a fix for this problem without getting my hands on an AMD GPU. I expect if they were doing the copy D3D12-side, SK would already be able to detect the format mismatch and correct it.

So it's either Vulkan-side or some proprietary thing in their driver that is exempt from validating image format :) I'm leaning heavily toward it's something in the driver I probably can't see. If it were being done D3D12-side and not accounted for, the format mismatch would be caught by the driver and the game would instantly crash.

**Tl;Dr:**

I think HDR is out of the question on AMD systems using Vulkan for now. Probably the only saving grace here is SK can enable HDR in D3D11 before DXVK translates it to Vulkan.

For native Vulkan software, I have no estimate if/when this will be fixed.

**Comment by SaltyBet** [comment]
Hi, @Kaldaien.

Over at https://github.com/GPUOpen-Drivers/AMD-Gfx-Drivers/issues/24#issuecomment-2384343026, AMD staff asked for the following:

> _From reading that thread it seems to be an issue in the vulkan application layer validating image format when translated from D3D12 SwapChain. Can you provide your system specs and a minimally compilable code source that reproduces this issue?_

I provided them my specs and referenced the [latest version of the SK source af of right now](https://github.com/SpecialKO/SpecialK/archive/refs/tags/SK_24_9_26.zip), but maybe you'd like to provide more details over there.

**Comment by SaltyBet** [comment]
Comments from AMD staff:

> _Are you able to provide full runtime of the issue? That's the source for the SK dll that does the injection from application to driver. Which could be the source of the issue, meaning it's not driver related, rather an application issue requiring a SK developer to dig into further. It doesn't seem like there's evidence that it's a driver issue currently._

**Comment by Ilan-Y** [comment]
I just upgraded to AMD recently and was sad to find out about this :\
Is there no update or even a temporary fix?

**Comment by Ilan-Y** [comment]
After fiddling with it a lot, I somehow got HDR10 with tonemapping to work on RPCS3 (Vulkan) using ReShade (scRGB was problematic).
For some reason, when I activate MajorPainTheCactus's AutoHDR plugin on ReShade together with SK HDR, it fixes the output even though there is an error message in ReShade from the AutoHDR plugin saying that HDR is not supported.

The same approach, however, didn't work on ShadPS4 (Vulkan), but I did manage to get ShadPS4 to output HDR10 without tonemapping without ReShade, but that's it. I had to go into windowed mode and fullscreen mode, and then back to windowed mode.

RPCS3 screenshots:
PNG:
![Image](https://github.com/user-attachments/assets/ca624a11-6ce6-4565-847b-9c9122598923)
I also have JXR screenshots you can view with an HDR image viewer:
https://drive.google.com/file/d/1Lx9Hl9vwrk8w0RULcSzXIvs0o1gRzg3o/view?usp=sharing

ShadPS4 screenshots:
PNG:
![Image](https://github.com/user-attachments/assets/5869ce82-c3e5-41fc-ac19-d9c95a04c617)
JXR:
https://drive.google.com/file/d/1gSS6f_psw-3wwBBo-Tko6TH5G28M9PPj/view?usp=sharing

I have no idea if this helps with creating a workaround or if anyone can benefit from this info.

**Comment by piqey** [comment]
@Ilan-Y Don't know if it helps you in particular but it may help someone—I've gotten scRGB HDR working for Bloodborne with ReShade's AutoHDR plugin _and_ RTX HDR together. The tonemapping is all messed up with ReShade AutoHDR alone and I couldn't fix it after actual hours of debugging, but RTX HDR somehow corrects it to a point where I can tonemap it to my liking.
